app/api/utils/xis_helper_functions.py

```diff
@@ -0,0 +1,68 @@
+import requests
+from configurations.models import XMSConfigurations
+
+
+# helper function to get the catalogs from XIS
+def get_xis_catalogs():
+    """Get all the catalogs available in the XIS
+
+    Returns:
+        requests.Response: [dictionary]
+    """
+
+    # get the XMS configuration for the XIS catalog host
+    xis_catalogs_url = (
+        XMSConfigurations.objects.first().target_xis_catalogs_host
+    )
+
+    # request the catalogs from the XIS
+    return requests.get(xis_catalogs_url)
+
+
+# helper function to get an experience from XIS
+def get_xis_experience(provider_id, experience_id):
+    """
+    Get a specific experience from a specific catalog
+
+    Args:
+        provider_id (string): the query parameter for the catalog
+        experience_id (strint): the metadata_key_hash for the experience
+
+    Returns:
+        requests.Response: [dictionary]
+
+    Note:
+        Only returns one experience in an array if successful
+    """
+
+    # get the XIS host from the configuration
+    xis_metadata_experience_url = (
+        XMSConfigurations.objects.first().target_xis_metadata_host
+    )
+
+    # construct the url for the experience
+    xis_metadata_experience_url = (
+        xis_metadata_experience_url
+        + f"?provider_id={provider_id}&metadata_key_hash_list={experience_id}"
```

Comment:
This doesn't work with the current XIS API.

This is used to retrieve a single experience, so provider_id shouldn't be included, and metadata_key_hash_list should be renamed **id**

Reply:
My mistake, I was using the unique_record_identifier instead of the metadata_key_hash.  You were correct